src/httpserver.cpp

```diff
@@ -480,6 +475,14 @@ void StopHTTPServer()
         workQueue->WaitExit();
         delete workQueue;
     }
+    if (eventBase) {
+        LogPrint("http", "Waiting for HTTP event thread to exit\n");
+        // Give event loop a few seconds to exit (to send back last RPC responses), then break it
+        if (!threadHTTP.try_join_for(boost::chrono::milliseconds(2000))) {
```

Comment:
- Waiting for all events to finish could take forever. There's tons of ways that someone could keep a connection option by slowly sending data. Also, timing events (such as the `walletpassphrase` timeout) may hold up the event queue.

- Just breaking off everything in progress immediately prevents the answer to 'stop' from getting back.

For context read #6719.

I'm not 100% happy with this solution either, but I wasn't able to find another way to do it, and it's better than the previous one.